dev/NavigationView/NavigationView.cpp

```diff
@@ -2161,8 +2161,12 @@ void NavigationView::OnNavigationViewItemTapped(const winrt::IInspectable& sende
 
 void NavigationView::OnNavigationViewItemKeyUp(const winrt::IInspectable& sender, const winrt::KeyRoutedEventArgs& args)
 {
-    // Because ListViewItem eats the events, we only get these keys on KeyUp.
-    if (args.OriginalKey() == winrt::VirtualKey::GamepadA)
+    // Because ListViewItem eats the events, we only get these keys on KeyUp AND
+    // If we handle space and enter upon initial key down, user can hold down and items get invoked rapidly
+    // That creates stuttering and we don't want that to happen
+    if (args.OriginalKey() == winrt::VirtualKey::GamepadA
```

Comment:
I am not sure whether we want to change the behavior to having selection change on KeyUp.
Maybe @chigy can weigh in?

Reply:
Is there a way to implement this without this? Would this impact all selection in the same way (select using any arrow key)?

Reply:
Key up/down issue is an age long debate that we should not be changing just looking at a single control. Given we do not have the cycles to visit the entire key up/down issue holistically, it is desirable that existing behavior not being modified.

Reply:
> Is there a way to implement this without this? Would this impact all selection in the same way (select using any arrow key)?

Unfortunately, there is not a way to fix the issue without intervening in the event. This does not affect arrow navigation etc.

> Key up/down issue is an age long debate that we should not be changing just looking at a single control. Given we do not have the cycles to visit the entire key up/down issue holistically, it is desirable that existing behavior not being modified.

So we should handle it on KeyDown then? Note that this would require additional logic to track if the event was raised after an KeyDown (repeated invoke) or KeyUp (valid invoke).

Reply:
@predavid , I wonder if we encountered the same or similar issues with TreeView and how we handled this?

Reply:
@chingucoding I think you could have a boolean flag on the nav view item that is set to true on key down and blocks selection and is set to false on key up. this way we still select on key down but prevent reactivation until the key up.

Reply:
Yes, that would work, but would still increase complexity (slightly). But let's wait until we decided on where to handle this.

Something that we need to consider regarding KeyUp/KeyDown: 
- Button, AppBarButton, CommandBar all raise invoked on KeyUp.
- TreeView handles it on KeyDown however also suffers from the problem that ItemInvoked gets raised repeatedly when space is being hold down.

Reply:
> Given we do not have the cycles to visit the entire key up/down issue holistically, it is desirable that existing behavior not being modified.

I agree that for now we should leave the existing behavior unmodified. 

So for now I think it makes most sense to move forward with the solution proposed by Stephen above.

Reply:
I agree with Dilip agreeing with myself :)

Reply:
Great, I'll adjust the code when I have time.